### PR TITLE
ENH: Subject hierarchy speed up with batch processing

### PR DESCRIPTION
**Pull request by cpinter**

### CODE REVIEW

**Comment by cpinter** [comment]
I'll integrate this tomorrow so that we have a preview version with https://github.com/Slicer/Slicer/pull/1215 and without this change. If there are any issues let me know.

**Comment by cpinter** [comment]
I found some issues, made some fixes, and it looks like this:
https://1drv.ms/v/s!Ao_a-dPPX98Zg4cyTpIHAN-GHEUolg?e=6OgxgD
@jcfr @pieper @fedorov @rkikinis Check out the video, I think you'll find it cool.

The main problem as always was after scene closing, that's why I included it in the video.

After show/hiding a big branch there some flickering and reseting of the layout spacer. Not sure why it happens, maybe something happens with the scene views after batch processing ends? @lassoan 
Otherwise (and even including this flickering :) ) it looks great to me. If there are no comments I'll integrate this tomorrow.

**Comment by jcfr** [comment]
> I found some issues, made some fixes, and it looks like this:
> https://1drv.ms/v/s!Ao_a-dPPX98Zg4cyTpIHAN-GHEUolg?e=6OgxgD
> Check out the video, I think you'll find it cool.

Well done :pray: 

And it is quite fast to toggle entire sub-hierarchy

File: Libs/MRML/Core/vtkMRMLFolderDisplayNode.cxx [hunk 1]
```diff
@@ -198,6 +199,13 @@ void vtkMRMLFolderDisplayNode::ChildDisplayNodesModified()
   // Get items in branch
   std::vector<vtkIdType> childItemIDs;
   shNode->GetItemChildren(folderItemId, childItemIDs, true);
+
+  bool batchProcessing = (childItemIDs.size() > 10);
```
Comment (jcfr) [inline Libs/MRML/Core/vtkMRMLFolderDisplayNode.cxx]: What would be the cons of always do batch processing ?

File: Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyModel.cxx [hunk 2]
```diff
@@ -757,6 +757,50 @@ void qMRMLSubjectHierarchyModel::updateFromSubjectHierarchy()
   emit subjectHierarchyUpdated();
 }
 
+//------------------------------------------------------------------------------
+void qMRMLSubjectHierarchyModel::updateFromSubjectHierarchy()
+{
+  Q_D(qMRMLSubjectHierarchyModel);
+
```
Comment (jcfr) [inline Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyModel.cxx]: Could you add comment to help understand the different condition ?

File: Libs/MRML/Core/vtkMRMLFolderDisplayNode.cxx (hunk 1 above)
Comment (cpinter) [reply Libs/MRML/Core/vtkMRMLFolderDisplayNode.cxx]: There is some overhead for doing batch processing. This number is supposed to compensate for that.

**Comment by cpinter** [comment]
Integrated in https://github.com/Slicer/Slicer/commit/e264333b443803a2afa8f563ff0fe373f52afa1b

**Pull request closed without merging**